osbs/tekton.py

```diff
@@ -590,180 +355,357 @@ def log_and_sleep():
+        )
+        r = self.os._get(url)
+        return r.json()
+
+    def wait_for_start(self):
+        logger.info("Waiting for pipeline run '%s' to start", self.pipeline_run_name)
+        for changetype, pipeline_run in self.os.watch_resource(self.api_path, self.api_version, resource_type='pipelineruns', resource_name=self.pipeline_run_name):
             try:
-                obj_status = obj["status"]["phase"]
+                status = pipeline_run['status']['conditions'][0]['status']
+                reason = pipeline_run['status']['conditions'][0]['reason']
             except KeyError:
-                logger.error("'object' doesn't have any status")
+                logger.error("pipeline run does not have any status")
                 continue
+            # pipeline run finished succesfully or failed
+            if status in ['True', 'False']:
+                return pipeline_run
+            elif status == 'Unknown' and reason == 'Running':
+                return pipeline_run
             else:
-                obj_status_lower = obj_status.lower()
-            logger.info("object has changed: '%s', status: '%s', name: '%s'",
-                        changetype, obj_status, obj_name)
-            if obj_name == build_id:
-                logger.info("matching build found")
-                logger.debug("is %s in %s?", repr(obj_status_lower), states)
-                if obj_status_lower in states:
-                    logger.debug("Yes, build is in the state I'm waiting for.")
-                    return obj
-                else:
-                    logger.debug("No, build is not in the state I'm "
-                                 "waiting for.")
-            else:
-                logger.info("The build %r isn't me %r", obj_name, build_id)
-
-        # I'm not sure how we can end up here since there are two possible scenarios:
-        #   1. our object was found and we are returning in the loop
-        #   2. our object was not found and we keep waiting (in the loop)
-        # Therefore, let's raise here
-        logger.warning("build '%s' was not found during wait", build_id)
-        raise OsbsWatchBuildNotFound("build '%s' was not found and response stream ended" %
-                                     build_id)
-
-    def wait_for_build_to_finish(self, build_id):
-        for retry in range(WAIT_RETRY):
-            try:
-                build_response = self.wait(build_id, BUILD_FINISHED_STATES)
-                return build_response
-            except OsbsWatchBuildNotFound:
-                # this is woraround for https://github.com/openshift/origin/issues/2348
-                logger.warning("I'm going to wait again. Retry #%d.", retry)
-                continue
-        raise OsbsException("Failed to wait for a build: %s" % build_id)
+                # (Unknown, Started), (Unknown, PipelineRunCancelled)
+                logger.debug("Waiting for pipeline run, current status %s, reason %s", status, reason)
```

Comment:
It's a bit confusing that Started doesn't mean started (at least according to this method). Could that be explained in a comment or docstring? Or maybe the method name needs to change?

Reply:
Also, what should happen if the pipeline doesn't start? Can the last status be something other than 'True', 'False'? The method would just implicitly return None in that case

Reply:
For your second comment - I think the last status cannot be anything other than True or False. So I am expecting that if the start pipeline code returns 200 OK status code, then the pipeline will have one of the statuses mentioned, Unknown, True or False.

Reply:
Ack, the table here pretty much answered both my questions -> https://tekton.dev/docs/pipelines/pipelineruns/#monitoring-execution-status

I'm fine with keeping the name, maybe just include that link as an explanation

Reply:
kept the name, added doc link to method docstring